lib/request-fragment.js

```diff
@@ -14,35 +13,37 @@ const requiredHeaders = {
  *  - filtered headers
  *  - Specified timeout from fragment attributes
  *
+ * @param {filterHeaders} - Function that handles the header forwarding
  * @param {string} fragmentUrl - URL of the fragment server
  * @param {Object} fragmentAttributes - Attributes passed via fragment tags
  * @param {Object} request - HTTP request stream
  * @returns {Promise} Response from the fragment server
  */
-module.exports = (fragmentUrl, fragmentAttributes, request) =>
-    new Promise((resolve, reject) => {
-        const parsedUrl = url.parse(fragmentUrl);
-        const options = Object.assign({
-            headers: Object.assign(
-                filterHeaders(fragmentAttributes, request.headers),
-                requiredHeaders
-            ),
-            keepAlive: true,
-            timeout: fragmentAttributes.timeout
-        }, parsedUrl);
-        const { protocol: reqProtocol, timeout } = options;
-        const protocol = reqProtocol === 'https:' ? https : http;
-        const fragmentRequest = protocol.request(options);
-        if (timeout) {
-            fragmentRequest.setTimeout(timeout, fragmentRequest.abort);
-        }
-        fragmentRequest.on('response', (response) => {
-            if (response.statusCode >= 500) {
-                reject(new Error('Internal Server Error'));
-            } else {
-                resolve(response);
+module.exports = (filterHeaders = () => ({})) =>
```

Comment:
is it worth putting a default argument here? what's the value? I'd prefer if something was wrong that it fails straight away - ie, throws if it receives undefined for whatever reason.